### PR TITLE
feat(storybook): add browser settings in the storybook

### PR DESCRIPTION
**Pull request by jrainville**
Adds browser settings to the storybook.

~~Branched off of https://github.com/status-im/status-desktop/pull/7944 so I could inspire myself~~

image

### CODE REVIEW

**Review by caybro — COMMENTED** [review]
Looks good! Some minor stuff only

**Comment by jrainville** [comment]
Great suggestions @caybro

File: ui/app/AppLayouts/Profile/views/BrowserView.qml [hunk 1]
```diff
@@ -76,12 +80,8 @@ SettingsContentBase {
...
                 components: [
                     StatusSwitch {
-                        checked: localAccountSensitiveSettings.shouldShowFavoritesBar
-                        onCheckedChanged: {
-                            if (localAccountSensitiveSettings.shouldShowFavoritesBar !== checked) {
-                                localAccountSensitiveSettings.shouldShowFavoritesBar = checked
-                            }
-                        }
+                        checked: accountSettings.shouldShowFavoritesBar
+                        onToggled: { accountSettings.shouldShowFavoritesBar = checked }
```
Comment (borismelnik) [inline ui/app/AppLayouts/Profile/views/BrowserView.qml:84]: I'm not sure if `checked` property was changed before `toggled` signal, so basically, its equal to `accountSettings.shouldShowFavoritesBar = accountSettings.shouldShowFavoritesBar`

**Review by micieslak — CHANGES_REQUESTED** [review]
Hey, I'm really glad to see storybook in use! :)

It's nice work however I have some doubts.
The page should run without any errors/warnings, but in this case I see many many of them:
```
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
file:///home/michalc/workspace/status-desktop/ui/StatusQ/src/StatusQ/Components/StatusListItem.qml:165:9: QML QQuickItem*: Binding loop detected for property "height"
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/views/wallet/PermissionsListView.qml:50:21: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/views/wallet/PermissionsListView.qml:50:21: Unable to assign [undefined] to int
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:78: TypeError: Cannot read property 'communitiesModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:77: TypeError: Cannot read property 'model' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:76: ReferenceError: communitiesModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:74: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:73: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Chat/stores/StickersStore.qml:9: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:68: ReferenceError: stickersModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:65: TypeError: Cannot read property 'keycardModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:57: ReferenceError: dappPermissionsModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:43: ReferenceError: walletSectionCurrent is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:26: ReferenceError: walletSectionAllTokens is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:25: ReferenceError: walletSectionAllTokens is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:24: ReferenceError: walletSectionAllTokens is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:23: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:22: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:21: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:20: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:14: ReferenceError: networksModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:13: ReferenceError: networksModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:12: ReferenceError: networksModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:11: ReferenceError: networksModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:9: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/EnsUsernamesStore.qml:17: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/EnsUsernamesStore.qml:15: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/EnsUsernamesStore.qml:13: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/EnsUsernamesStore.qml:12: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/EnsUsernamesStore.qml:11: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:58: TypeError: Cannot read property 'ensUsernamesModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/PrivacyStore.qml:10: TypeError: Cannot read property 'mnemonicBackedUp' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:54: TypeError: Cannot read property 'privacyModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:548: ReferenceError: mainModule is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/utils/Utils.qml:10: ReferenceError: globalUtils is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:23: TypeError: Cannot read property 'socialLinksDirty' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:22: TypeError: Cannot read property 'temporarySocialLinksModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:21: TypeError: Cannot read property 'socialLinksModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:20: TypeError: Cannot read property 'socialLinksJson' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:19: TypeError: Cannot read property 'bio' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:17: ReferenceError: profileSectionModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:16: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:15: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:14: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:13: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:12: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:11: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:10: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileStore.qml:9: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:50: TypeError: Cannot read property 'profileModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/LanguageStore.qml:12: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/LanguageStore.qml:11: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:43: TypeError: Cannot read property 'languageModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:59: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:55: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:51: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:47: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:43: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:39: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:35: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:31: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:27: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:23: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:19: ReferenceError: appSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/NotificationsStore.qml:9: TypeError: Cannot read property 'exemptionsModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:39: TypeError: Cannot read property 'notificationsModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/DevicesStore.qml:12: TypeError: Cannot read property 'isDeviceSetup' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/DevicesStore.qml:9: TypeError: Cannot read property 'model' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:35: TypeError: Cannot read property 'devicesModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/MessagingStore.qml:20: TypeError: Cannot read property 'activeMailserver' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/MessagingStore.qml:19: TypeError: Cannot read property 'automaticSelection' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/MessagingStore.qml:12: TypeError: Cannot read property 'useMailservers' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/MessagingStore.qml:10: TypeError: Cannot read property 'model' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:31: TypeError: Cannot read property 'syncModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:30: TypeError: Cannot read property 'privacyModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:26: TypeError: Cannot read property 'advancedModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:17: TypeError: Cannot read property 'sentContactRequestsModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:16: TypeError: Cannot read property 'receivedContactRequestsModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:15: TypeError: Cannot read property 'blockedContactsModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:14: TypeError: Cannot read property 'myMutualContactsModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:12: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:10: ReferenceError: mainModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ContactsStore.qml:9: ReferenceError: globalUtils is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:22: TypeError: Cannot read property 'contactsModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:19: ReferenceError: aboutModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:17: ReferenceError: profileSectionModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:15: ReferenceError: aboutModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:13: TypeError: Cannot read property 'unreadCount' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:12: TypeError: Cannot read property 'activityNotificationsModel' of undefined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/ProfileSectionStore.qml:11: ReferenceError: activityCenterModule is not defined
file:///home/michalc/workspace/status-desktop/ui/app/AppLayouts/Profile/stores/WalletStore.qml:43: ReferenceError: walletSectionCurrent is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:114: ReferenceError: chatSectionChatContentInputArea is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:113: ReferenceError: chatSectionChatContentInputArea is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:112: ReferenceError: chatSectionChatContentInputArea is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/TokenMarketValuesStore.qml:55: TypeError: Cannot read property 'length' of undefined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:44: ReferenceError: walletSectionAccounts is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:43: ReferenceError: walletSectionAllTokens is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:42: ReferenceError: walletSectionAllTokens is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:41: ReferenceError: walletSectionCurrent is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:39: TypeError: Cannot read property 'isNonArchivalNode' of undefined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:38: ReferenceError: walletSectionTransactions is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:37: ReferenceError: walletSectionTransactions is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:32: ReferenceError: walletSection is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/CurrenciesStore.qml:7: ReferenceError: walletSection is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:17: ReferenceError: localAccountSensitiveSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:16: ReferenceError: localAppSettings is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:15: ReferenceError: walletSection is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:14: ReferenceError: userProfile is not defined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:13: TypeError: Cannot read property 'privacyModule' of undefined
file:///home/michalc/workspace/status-desktop/ui/imports/shared/stores/RootStore.qml:12: ReferenceError: profileSectionModule is not defined
```
The reason seems to be the fact that real stores are used instead of the mocked one. Please check https://github.com/status-im/status-desktop/blob/master/storybook/pages/CommunitiesPortalLayoutPage.qml for reference how to use fully mocked stores by adding stub in `stubs` dir and implementing mocked login in a page file.

**Review by borismelnik — APPROVED** [review]

File: storybook/pages/BrowserSettingsPage.qml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+                    generatedAccounts: null
+                    watchOnlyAccounts: null
+                    walletTokensModule: null
+                    defaultTokenList: null
+                    customTokenList: null
+
+                    function disconnect(dappName) {
+                        for (let i = 0; i < dappsModel.count; i++) {
+                            if (dappsModel.get(i).name === dappName) {
+                                dappsModel.remove(0)
```
[the remark this answers is not in the document]
Comment (jrainville) [reply storybook/pages/BrowserSettingsPage.qml:79]: No :see_no_evil:

File: storybook/pages/BrowserSettingsPage.qml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+SplitView {
+    Logs { id: logs }
+
+    ListModel {
+        id: dappsModel
+
+        ListElement {
+            name: "http://simpledapp.eth"
+            // TODO find a way to imbricate another ListModel for the accounts
```
[the remark this answers is not in the document]
Comment (jrainville) [reply storybook/pages/BrowserSettingsPage.qml:20]: That was fixed in a later version. I just forgot to remove the comment sorry.

**Comment by jrainville** [comment]
@micieslak thanks for the review. I updated accordingly

**Review by micieslak — CHANGES_REQUESTED** [review]
Thanks for fixes, really good job! Some other smaller issues found, but now it's in a much better shape.

At the startup, there is a warning:
```
WebEngineContext used before QtWebEngine::initialize() or OpenGL context creation failed.
```
We need to add `WebEngine` to `CMakeLists.txt` and call `QtWebEngine::initialize();` in main to fix it.

File: storybook/pages/BrowserSettingsPage.qml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+                    color: "#887AF9"
+                    emoji: "😋"
+                }
+            ]
+        }
+    }
+
+    property QtObject mockData: QtObject {
+        property var accountSettings: QtObject {
+            property string browserHomepage: "https://status.im/"
```
[the remark this answers is not in the document]
Comment (jrainville) [reply storybook/pages/BrowserSettingsPage.qml:39]: We use QtProperties for Settings, so technically, in the backend of the real data, it does use a setter. Changing all settings to use the setters directly and also adding those setters to the mock data sounds very painful for the little gain it adds a far as I see it.
What do you think?

**Comment by jrainville** [comment]
@micieslak updated!

File: storybook/pages/BrowserSettingsPage.qml (hunk 4 above)
[the remark this answers is not in the document]
Comment (micieslak) [reply storybook/pages/BrowserSettingsPage.qml:39]: for now I would leave it as it is, but for the future we should aim modifying state of backend via methods exposed from stores. And expose state of backed via readonly properties from stores. It's really beneficial from the code design perspective.

**Review by micieslak — APPROVED** [review]
cool!

**Pull request merged**